### DISCUSSION TITLE
How to increase NSM storage in a distributed environment?

### DISCUSSION THREAD

**Discussion by loganterfehr** [Security-Onion-Solutions/securityonion#12270] [Unsupported Versions] [answered]
I have a distributed environment that is in vCenter VMs. I increased the storage size of the drive and then partitioned it to sda3 with the remaining free space using 'cfdisk' command. However, is this not all that I have to do? The status does not seem to update to the new amount for nsm. Am I doing this right?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/a9878b11-3228-4612-b356-4f96fdb399db)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/4dc5e201-3cb4-40f0-aa5f-9e4d7ba93bd9)
I am supposed to put more storage in sda3 right? The LVM

**Comment by cm-ops** [comment]
How about `df -h` command, what does that show? If you added space to your `/nsm` partition you will probably need to still grow the filesystem.

**Reply by loganterfehr** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/122332360/3dac651e-6cfb-4e9e-a604-d620003c755a)
This is what that looks like. A little confused where the space went that I allocated to the LVM drive since that space shows up nowhere when I run `df -h`. Thanks for helping me with this.

**Reply by cm-ops** [reply]
If you have the space allocated to the `/nsm` partion, run `xfs_growfs /dev/mapper/system-nsm` to grow the filesystem.

**Reply by loganterfehr** [reply]
Okay I think I am a little confused. Would you be willing to give a step by step? I freed some space that is not allocated to anything anymore. What steps should I take to get /nsm partitioned. Thanks Chris.

**Reply by petiepooo** [reply, marked as answer]
There are layers.  You always need to work from the lowest layer upwards...  
If you have increased the size of the device presented to the VM, it should show up in your VM at the device level.  Try the command `sudo blockdev --getsize64 /dev/sda` to see the raw byte count.  If it hasn't increased yet, try rescanning the SCSI bus using the command `echo 1 | sudo tee /sys/class/block/sda/device/rescan`.
If the new space is not yet used in a partition as shown in `sudo parted /dev/sda print`, increase the size of the last partition (in your case partition 3) with the command `sudo growpart /dev/sda 3`.  
If `sudo pvs` still shows PV /dev/sda3 as your old size, you need to increase it with `sudo pvresize /dev/sda3` (it defaults to use all available space).  
If `sudo lvs` still shows your old size, you need to increase it with `sudo lvextend -l +100%FREE /dev/mapper/system-nsm` (that will use all available space in the volume group).  
Finally grow the filesystem on the LV.  For xfs filesystems, use `sudo xfs_growfs /dev/mapper/system-nsm`.  For ext4, use `sudo resize2fs /dev/mapper/system-nsm`.  Both those default to using the full device if given no other options.

**Reply by loganterfehr** [reply]
That was extremely well written and described. Thanks a ton Pete and Chris!
